File: sdks/javascript/examples/poll-status.js
// Exemple : polling du statut d'une vérification existante
const Datakeys = require('@datakeys/kyc').default;

const dk = new Datakeys('dk_test_datakeys_sandbox_001');
const verificationId = process.argv[2];

if (!verificationId) {
  console.error('Usage: node poll-status.js <verification_id>');
  process.exit(1);
}

async function main() {
  const result = await dk.kyc.waitForCompletion(verificationId, {
    maxWaitMs: 120_000,
    intervalMs: 3_000,
    onPoll: (v) => {
      const elapsed = Math.round((Date.now() - v.created * 1000) / 1000);
      console.log(`[${elapsed}s] Status: ${v.status}  Score: ${v.score}`);
    },
  });

  console.log('\n=== RÉSULTAT FINAL ===');
  console.log(JSON.stringify(result, null, 2));
}

main().catch((err) => {
  console.error('Erreur:', err.message);
  process.exit(1);
});

File: sdks/python/examples/poll_status.py
"""Exemple : polling du statut d'une vérification existante"""
import sys
import time
from datakeys import Datakeys

dk = Datakeys("dk_test_datakeys_sandbox_001")

if len(sys.argv) < 2:
    print("Usage: python poll_status.py <verification_id>")
    sys.exit(1)

verification_id = sys.argv[1]


def main():
    start = time.time()
    result = dk.kyc.wait_for_completion(
        verification_id,
        max_wait=120,
        interval=3,
        on_poll=lambda v: print(
            f"[{int(time.time() - start)}s] Status: {v.status}  Score: {v.score}"
        ),
    )

    print("\n=== RÉSULTAT FINAL ===")
    print(f"Status: {result.status}")
    print(f"Score:  {result.score}")


if __name__ == "__main__":
    try:
        main()
    except Exception as e:
        print(f"Erreur: {e}")
        sys.exit(1)
